### PR TITLE
Brendan/#34 profile foundations

### LINKED ISSUE #34: Profile Microservice Foundations
https://trello.com/c/dupP22Mk/16-profile-microservice-foundations

### PR DESCRIPTION
**Pull request by Silver-Golden**
Not 100% sure that this is correct, I have a rough idea how it all *should* work but not about the specifics.  
Core function behind this is getting and setting profile data.  
Last time I checked I was able to build it, but running it was failing on accessing teh database.  
I will create a new issue to get a database config up and running, at least on a temp basis

Closes #34

### CODE REVIEW

File: Microservices/Profile/src/main/java/cs4337/group_8/TemplateProject/DTO/ProfileDTO.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+public class ProfileDTO {
+// string as it should be alphanumeric
+    private String user_id;
+    private String full_name;
+    private String bio;
+    private String profile_pic;
+    private int count_follower;
+    private int count_following;
+
+//    public ProfileDTO (String property1){
```
Comment (M-Byte480) [inline Microservices/Profile/src/main/java/cs4337/group_8/TemplateProject/DTO/ProfileDTO.java:20]: Can you remove this

File: Microservices/Profile/src/main/java/cs4337/group_8/TemplateProject/controllers/ProfileController2.java
Comment (M-Byte480) [inline Microservices/Profile/src/main/java/cs4337/group_8/TemplateProject/controllers/ProfileController2.java:1]: Can you rename it without the 2

Comment (M-Byte480) [inline Microservices/Profile/src/main/java/cs4337/group_8/TemplateProject/controllers/ProfileController2.java:1]: On that note, can you change the package name to profileMicroservice or similar (profilems)

File: Microservices/Profile/src/main/java/cs4337/group_8/TemplateProject/repositories/ProfileRepository.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+public interface ProfileRepository extends JpaRepository<ProfileEntity, String> {
+
+    // Optional is a container object which may or may not contain a non-null value
+    Optional<ProfileEntity> findByIdEquals(String id);
+//    Optional<List<ProfileEntity>> findAllByDateOfBirthBefore(Date date);
+//    Optional<List<ProfileEntity>> findAllByVerifiedTrue();
+
+    // You can also use the @Query annotation to write custom queries
+    //@Query(value = "SELECT e FROM ProfileEntity e WHERE e.firstname = ?1 AND e.verified = ?2")
+    // Where ?1 is the first parameter of the method
```
Comment (M-Byte480) [inline Microservices/Profile/src/main/java/cs4337/group_8/TemplateProject/repositories/ProfileRepository.java:23]: Can you remove all the comments

File: Microservices/Profile/src/main/java/cs4337/group_8/TemplateProject/services/ProfileService.java [hunk 3]
```diff
@@ -41,7 +42,7 @@ public List<SampleEntity> getAllUsersByName(String name){
             if (query.isPresent()){
                 return query.get();
             } else {
-                return new ArrayList<SampleEntity>();
+                return new ArrayList<ProfileEntity>();
             }
```
Comment (M-Byte480) [inline Microservices/Profile/src/main/java/cs4337/group_8/TemplateProject/services/ProfileService.java:46]: Commented code here

**Review by M-Byte480 — CHANGES_REQUESTED** [review]
Just a couple of nit picks
TODO: add unit tests

File: Microservices/Profile/src/main/java/cs4337/group_8/TemplateProject/DTO/ProfileDTO.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+//        this.property1 = property1;
+//    }
```
Comment (M-Byte480) [inline Microservices/Profile/src/main/java/cs4337/group_8/TemplateProject/DTO/ProfileDTO.java:22]: Remove the commented out code

File: Microservices/Profile/src/main/java/cs4337/group_8/TemplateProject/entities/ProfileEntity.java [hunk 5]
```diff
@@ -19,26 +19,34 @@
...
 @Entity
 @Data
 @ToString(callSuper = true, includeFieldNames = true)
-@Table(name = "name_of_table")
-public class SampleEntity {
+@Table(name = "profile")
+public class ProfileEntity {
     @Id // Primary key
-    @GeneratedValue(strategy = GenerationType.IDENTITY) // Auto-increment in the database
+//    @GeneratedValue(strategy = GenerationType.IDENTITY) // Auto-increment in the database
```
Comment (M-Byte480) [inline Microservices/Profile/src/main/java/cs4337/group_8/TemplateProject/entities/ProfileEntity.java:29]: Commented out generationPattern

File: Microservices/Profile/src/main/java/cs4337/group_8/TemplateProject/repositories/ProfileRepository.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    //Optional<List<ProfileEntity>> findAllByFirstnameAndVerified(String firstname, boolean verified);
```
Comment (M-Byte480) [inline Microservices/Profile/src/main/java/cs4337/group_8/TemplateProject/repositories/ProfileRepository.java:24]: Unused queries

File: Microservices/Profile/src/main/java/cs4337/group_8/TemplateProject/controllers/ProfileController2.java
Comment (M-Byte480) [inline Microservices/Profile/src/main/java/cs4337/group_8/TemplateProject/controllers/ProfileController2.java:1]: Can you remove the 2 and also rename the package from TemplateProject

File: Microservices/Profile/src/main/java/cs4337/group_8/TemplateProject/services/ProfileService.java [hunk 7]
```diff
@@ -41,7 +42,7 @@ public List<SampleEntity> getAllUsersByName(String name){
             if (query.isPresent()){
                 return query.get();
             } else {
-                return new ArrayList<SampleEntity>();
+                return new ArrayList<ProfileEntity>();
```
Comment (M-Byte480) [inline Microservices/Profile/src/main/java/cs4337/group_8/TemplateProject/services/ProfileService.java:45]: Commented code

**Review by M-Byte480 — CHANGES_REQUESTED** [review]

File: Microservices/Profile/src/main/java/cs4337/group_8/TemplateProject/controllers/ProfileController2.java
Comment (Silver-Golden) [reply Microservices/Profile/src/main/java/cs4337/group_8/TemplateProject/controllers/ProfileController2.java:1]: Funnily enough it failed to compile without it, kept claiming there was a conflict, even when I cleared the build cache, might end up naming it to be something else.

**Review by M-Byte480 — APPROVED** [review]
Good enough for starters

File: Microservices/Profile/pom.xml [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+	<modelVersion>4.0.0</modelVersion>
+
+	<parent>
+		<groupId>cs4337.group-8</groupId>
+		<artifactId>parent-microservices</artifactId>
+		<version>0.0.1-SNAPSHOT</version>
+		<relativePath>../pom.xml</relativePath>
+	</parent>
+
+	<artifactId>TemplateProject</artifactId>
```
Comment (Bourkupine) [inline Microservices/Profile/pom.xml:13]: Should this not be 'Profile'

File: Microservices/Profile/pom.xml [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+	<version>0.0.1-SNAPSHOT</version>
+	<name>TemplateProject</name>
```
Comment (Bourkupine) [inline Microservices/Profile/pom.xml:15]: same as above

**Review by Bourkupine — COMMENTED** [review]
Left comments

**Comment by M-Byte480** [comment]
@Bourkupine @faad555 @killianC2323 @silpac-12 Can you guys take a look, I registered the application onto Eureka

File: Microservices/service-registry/src/main/resources/application.properties [hunk 10]
```diff
@@ -8,13 +8,3 @@ eureka.instance.hostname=localhost
 eureka.client.fetch-registry=false
 eureka.client.register-with-eureka=false
```
[the remark this answers is not in the document]
Comment (Silver-Golden) [reply Microservices/service-registry/src/main/resources/application.properties:10]: Seems @M-Byte480 removed it in 0a6cb6a3250e18c2d212825accdbe1d89012f5e0

[the remark this answers is not in the document]
Comment (M-Byte480) [reply Microservices/service-registry/src/main/resources/application.properties:10]: Ye Service Registry is not a standalone api so it cant actually create and read DB. That is why each microservice will have their own db to read and write from

**Review by M-Byte480 — APPROVED** [review]

**Review by Bourkupine — APPROVED** [review]
Looks good to me, good work Brendan.

**Pull request merged**